### DISCUSSION TITLE
How to use JSDoc syntax with JS/TS

### DISCUSSION THREAD

**Discussion by trev-dev** [emacs-tree-sitter/elisp-tree-sitter#211] [Q&A]
I apologize for my ignorance in advance. I'm not up-to-snuff on how tree-sitter works and am struggling a bit with my Googling/reading to find a solution.

I can't seem to get JSDoc highlighting with tree-sitter in Emacs. The debugger just says the whole block is a `comment:`. I have `tree-sitter-languages` installed. I tried adding jsdoc to typescript mode in the languages alist and it broke tree-sitter for TS entirely.

I'm sure this might be some sort of configuration problem. Any insight would be appreciated.

**Comment by ubolonton** [comment]
`elisp-tree-sitter` [doesn't support multi-language buffers](#33) at the moment, so JSDoc fragments inside JS/TS are not recognized, since it's a separate `jsdoc` grammar.

**Reply by trev-dev** [reply]
Aaaah gotcha. But I'm assuming it could if someone knew enough about elisp and tree-sitter?

**Reply by ubolonton** [reply]
Yeah, I don't think this requires modifying the dynamic module part.
